project_timesheet_management/models/timesheet_invoice.py

```diff
@@ -0,0 +1,764 @@
+
+    @api.onchange("partner_id")
+    def _onchange_partner_id(self):
+        """Update the project value."""
+        self.project_id = False
+
+    def _compute_user_id_admin(self):
+        for invoice in self:
+            invoice.user_is_admin = self.env.user._is_admin()
+
+    def get_timesheet_invoice_url(self):
+        """
+        Generate Timesheet Invoice URL
+        """
+        self.ensure_one()
+        try:
+            base_url = self.env["ir.config_parameter"].get_param("web.base.url")
+            invoice_url = urls.url_join(
+                base_url, "web#id=%s&view_type=form&model=%s" % (self.id, self._name)
+            )
+        except Exception:
+            pass
+        return invoice_url
+
+    def button_draft(self):
+        """
+        Set timesheet invoice to draft.
+        """
+        self.with_context(from_approval=True).sudo().write({"state": "draft"})
+
+    def button_confirm(self):
+        """
+        Set timesheet invoice to confirm and send mail
+        Project Manager to Pre-Approve the Timesheet Invoice.
+        """
+        mail_template = self.env.ref(
+            "project_timesheet_management.email_timesheet_invoice_confirmed"
+        )
+        for invoice in self:
+            invoice.write({"state": "confirm"})
+            if mail_template:
+                email_from = self.env.user.email
+                email_to = invoice.project_id.user_id.email or ''
+                reply_to = self.env.user.email
+                mail_template.write({'email_from': email_from,
+                                     'email_to': email_to,
+                                     'reply_to': reply_to})
+                state = dict(invoice.fields_get(
+                    allfields=['state'])['state']['selection'])[invoice.state]
+                mail_template.sudo().with_context(state=state).send_mail(
+                    invoice.id, force_send=True
+                )
+
+    def button_pre_approve(self):
+        """
+        Set timesheet invoice to pre-approve and add the pre_approved_on
+        and send mail to CIO to Approve the Timesheet Invoice.
+        """
+        self._check_budget()
+        mail_template = self.env.ref(
+            "project_timesheet_management.email_timesheet_invoice_confirmed"
+        )
+        for invoice in self:
+            invoice.write(
+                {
+                    "pre_approved_on": datetime.now(),
+                    "pre_approved_by": self.env.user.id,
+                    "state": "pre-approved",
+                }
+            )
+            if mail_template:
+                email_from = self.env.user.email
+                email_to = invoice.project_id.product_owner_id.email or ''
+                reply_to = self.env.user.email
+                state = dict(invoice.fields_get(
+                    allfields=['state'])['state']['selection'])[invoice.state]
+                mail_template.write({'email_from': email_from,
```

Comment:
You should not write to the email template, instead you can use email_values as parameter for send_mail function